src/rhsm/connection.py

```diff
@@ -499,7 +499,10 @@ def _request(self, request_type, method, info=None, headers=None):
 
         if self.proxy_hostname and self.proxy_port:
             log.debug("Using proxy: %s:%s" % (self.proxy_hostname, self.proxy_port))
-            proxy_headers = {'User-Agent': self.user_agent}
+            proxy_headers = {
+                'User-Agent': self.user_agent,
+                'Host': '%s:%s' % (self.host, safe_int(self.ssl_port))
```

Comment:
Please normalize host for IPv6 compatibility as in #1725. (See https://github.com/candlepin/subscription-manager/blob/be328f04c089be2eccc508eba58391643ef07bfd/src/rhsm/connection.py#L327-L334)

Reply:
Sure thing!